# Quantitative Momentum Startegy
Investing in the sotcks that have increase in price most

#   Library import

In [126]:
import numpy as np
import pandas as pd
import requests 
import math
from scipy import stats
import xlsxwriter

# import Our List stocks

In [127]:
stocks = pd.read_csv("sp_500_stocks.csv")
stocks

Ticker
0        A
1      AAL
2      AAP
3     AAPL
4     ABBV
..     ...
500    YUM
501    ZBH
502   ZBRA
503   ZION
504    ZTS

[505 rows x 1 columns]

In [128]:
from secrets import IEX_CLOUD_API_TOKEN

# Making our First call

In [129]:
symbol = 'AAPL'
api_url = f'https://sandbox.iexapis.com/stable/stock/{symbol}/stats/?token={IEX_CLOUD_API_TOKEN}'
data  = requests.get(api_url).json()
data

{'companyName': 'Apple Inc',
 'marketcap': 2941854353138,
 'week52high': 189.65,
 'week52low': 115.75,
 'week52highSplitAdjustOnly': 187.76,
 'week52lowSplitAdjustOnly': 121.39,
 'week52change': 0.2724631234933916,
 'sharesOutstanding': 16401406753,
 'float': 0,
 'avg10Volume': 92067462,
 'avg30Volume': 103804119,
 'day200MovingAvg': 159.43,
 'day50MovingAvg': 175.79,
 'employees': 150954,
 'ttmEPS': 6.31,
 'ttmDividendRate': 0.8898855611258404,
 'dividendYield': 0.005293224559312313,
 'nextDividendDate': '',
 'exDividendDate': '2022-01-28',
 'nextEarningsDate': '2022-04-20',
 'peRatio': 29.16845294770324,
 'beta': 1.3726929989140095,
 'maxChangePercent': 65.89034263704504,
 'year5ChangePercent': 4.682824931478863,
 'year2ChangePercent': 1.172698620000062,
 'year1ChangePercent': 0.2715055591051016,
 'ytdChangePercent': -0.028889615990023,
 'month6ChangePercent': 0.18420951528485,
 'month3ChangePercent': 0.14380196173832582,
 'month1ChangePercent': 0.003651809252100125,
 'day30ChangePer

# Parising our API Call

In [130]:
data['year1ChangePercent']

0.2715055591051016

# Batch API call

In [131]:
def chunks(lst, n):
    """Yield successive n-sized chunks from lst."""
    for i in range(0, len(lst), n):
        yield lst[i:i + n]
        
symbol_group = list(chunks(stocks['Ticker'],100))
symbol_strings = []
for i in range(0,len(symbol_group)):
    symbol_strings.append(','.join(symbol_group[i]))        

my_columns = ['Ticker','Price','One-Year Price Return','Number of Share to Buy']

In [54]:
final_dataframe = pd.DataFrame(columns=my_columns)
for symbol_string in symbol_strings:
    batch_api_url = f'https://sandbox.iexapis.com/stable/stock/market/batch?symbols={symbol_string},fb&types=price,stats&token={IEX_CLOUD_API_TOKEN}'
#     print(batch_api_url)
    data = requests.get(batch_api_url).json()
    for symbol in (symbol_string.split(',')):
        try:
            final_dataframe = final_dataframe.append(
                pd.Series(
                    [
                        symbol,
                        data[symbol]['price'],
                        data[symbol]['stats']['year1ChangePercent'],
                        'N/A'
                    ],index = my_columns),
                ignore_index=True
            )
        except:
            print("This Symbol having issue while fetching data->"+str(symbol))
final_dataframe

This Symbol having issue while fetching data->WLTW


Ticker   Price One-Year Price Return Number of Share to Buy
0        A  143.36              0.154629                    N/A
1      AAL   17.10             -0.044791                    N/A
2      AAP  237.81              0.474898                    N/A
3     AAPL  176.89              0.277091                    N/A
4     ABBV  145.93              0.362559                    N/A
..     ...     ...                   ...                    ...
499    YUM  128.19              0.217306                    N/A
500    ZBH  125.30             -0.230688                    N/A
501   ZBRA  526.00              0.246208                    N/A
502   ZION   72.52              0.545853                    N/A
503    ZTS  208.68              0.268547                    N/A

[504 rows x 4 columns]

# Removing Low Momemtum stock

In [132]:
final_dataframe.sort_values('One-Year Price Return',ascending=False,inplace=True )
final_dataframe=  final_dataframe[:50]
final_dataframe.reset_index(inplace=True)
final_dataframe

C:\Users\aakash\Anaconda3\lib\site-packages\pandas\util\_decorators.py:311: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return func(*args, **kwargs)


level_0  index Ticker    Price One-Year Price Return  \
0         0    275     LB    80.36              2.375547   
1         1    148    DVN    54.30              2.118189   
2         2    317    MRO    22.50               1.78459   
3         3    109    COP    93.08              1.187643   
4         4    344    NUE   116.49              1.158484   
5         5    160    EOG   116.50              1.144546   
6         6    175   FANG   138.31              1.112231   
7         7    299   MCHP    73.47              1.073055   
8         8    195   FTNT   314.52               1.05686   
9         9    386    PXD   231.73              0.882716   
10       10    357    OXY    41.40              0.834849   
11       11    251     IT   300.22              0.829612   
12       12     89     CF    75.92              0.815042   
13       13    345   NVDA   246.93              0.812374   
14       14     42    APA    35.75              0.804569   
15       15    173    EXR   209.43              0.796876   
16       16    482    WFC    58.44              0.773898   
17       17    495    XOM    83.65              0.740079   
18       18    395    RHI   119.38              0.720571   
19       19    315    MPC    81.26              0.713734   
20       20    218    HES    99.95               0.70543   
21       21    212    HAL    32.49              0.693694   
22       22    407   SCHW    93.50              0.685253   
23       23     54    AZO  2024.90              0.668279   
24       24    382    PSA   375.17              0.656602   
25       25    295    MAA   213.49              0.644294   
26       26     98    CMA    97.93              0.626541   
27       27    314    MOS    45.30              0.598489   
28       28    365    PFE    55.00              0.592542   
29       29    174      F    18.55              0.588859   
30       30    352    OKE    65.56              0.582043   
31       31    123    CVX   136.60              0.580966   
32       32    417    SPG   149.53              0.579141   
33       33     39   ANTM   451.62               0.57747   
34       34    465    VAR   178.59              0.573645   
35       35    412    SLB    39.28              0.570921   
36       36    122    CVS   108.96              0.561763   
37       37    396    RJF   116.34              0.559671   
38       38    381    PRU   122.45              0.553847   
39       39    494   XLNX   218.15              0.546497   
40       40    502   ZION    72.52              0.545853   
41       41     56    BAC    50.49              0.539984   
42       42    185   FITB    49.48              0.538111   
43       43     37   ANET   128.40              0.526914   
44       44     82   CBRE   102.60               0.52228   
45       45     64     BK    63.68              0.518721   
46       46    458    UNH   486.15              0.518599   
47       47    468    VLO    90.90              0.515796   
48       48    273    KSU   294.58              0.510877   
49       49    249    IRM    45.92              0.505446   

   Number of Share to Buy  
0                    2488  
1                    3683  
2                    8888  
3                    2148  
4                    1716  
5                    1716  
6                    1446  
7                    2722  
8                     635  
9                     863  
10                   4830  
11                    666  
12                   2634  
13                    809  
14                   5594  
15                    954  
16                   3422  
17                   2390  
18                   1675  
19                   2461  
20                   2001  
21                   6155  
22                   2139  
23                     98  
24                    533  
25                    936  
26                   2042  
27                   4415  
28                   3636  
29                  10781  
30                   3050  
31                   1464  
32                 

# Calulating Number of Share to buy

In [133]:
portfoliosize = float(10000000)
postion_size = portfoliosize/len(final_dataframe)
postion_size
for i in range(0,len(final_dataframe)):
    final_dataframe.loc[i,'Number of Share to Buy'] = math.floor(postion_size/final_dataframe.loc[i,'Price'])
final_dataframe

C:\Users\aakash\Anaconda3\lib\site-packages\pandas\core\indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


level_0  index Ticker    Price One-Year Price Return  \
0         0    275     LB    80.36              2.375547   
1         1    148    DVN    54.30              2.118189   
2         2    317    MRO    22.50               1.78459   
3         3    109    COP    93.08              1.187643   
4         4    344    NUE   116.49              1.158484   
5         5    160    EOG   116.50              1.144546   
6         6    175   FANG   138.31              1.112231   
7         7    299   MCHP    73.47              1.073055   
8         8    195   FTNT   314.52               1.05686   
9         9    386    PXD   231.73              0.882716   
10       10    357    OXY    41.40              0.834849   
11       11    251     IT   300.22              0.829612   
12       12     89     CF    75.92              0.815042   
13       13    345   NVDA   246.93              0.812374   
14       14     42    APA    35.75              0.804569   
15       15    173    EXR   209.43              0.796876   
16       16    482    WFC    58.44              0.773898   
17       17    495    XOM    83.65              0.740079   
18       18    395    RHI   119.38              0.720571   
19       19    315    MPC    81.26              0.713734   
20       20    218    HES    99.95               0.70543   
21       21    212    HAL    32.49              0.693694   
22       22    407   SCHW    93.50              0.685253   
23       23     54    AZO  2024.90              0.668279   
24       24    382    PSA   375.17              0.656602   
25       25    295    MAA   213.49              0.644294   
26       26     98    CMA    97.93              0.626541   
27       27    314    MOS    45.30              0.598489   
28       28    365    PFE    55.00              0.592542   
29       29    174      F    18.55              0.588859   
30       30    352    OKE    65.56              0.582043   
31       31    123    CVX   136.60              0.580966   
32       32    417    SPG   149.53              0.579141   
33       33     39   ANTM   451.62               0.57747   
34       34    465    VAR   178.59              0.573645   
35       35    412    SLB    39.28              0.570921   
36       36    122    CVS   108.96              0.561763   
37       37    396    RJF   116.34              0.559671   
38       38    381    PRU   122.45              0.553847   
39       39    494   XLNX   218.15              0.546497   
40       40    502   ZION    72.52              0.545853   
41       41     56    BAC    50.49              0.539984   
42       42    185   FITB    49.48              0.538111   
43       43     37   ANET   128.40              0.526914   
44       44     82   CBRE   102.60               0.52228   
45       45     64     BK    63.68              0.518721   
46       46    458    UNH   486.15              0.518599   
47       47    468    VLO    90.90              0.515796   
48       48    273    KSU   294.58              0.510877   
49       49    249    IRM    45.92              0.505446   

   Number of Share to Buy  
0                    2488  
1                    3683  
2                    8888  
3                    2148  
4                    1716  
5                    1716  
6                    1446  
7                    2722  
8                     635  
9                     863  
10                   4830  
11                    666  
12                   2634  
13                    809  
14                   5594  
15                    954  
16                   3422  
17                   2390  
18                   1675  
19                   2461  
20                   2001  
21                   6155  
22                   2139  
23                     98  
24                    533  
25                    936  
26                   2042  
27                   4415  
28                   3636  
29                  10781  
30                   3050  
31                   1464  
32                 

# Build a Better Momemtum Startegy

Finding High Momemtum 

In [184]:
hqm_columns =[
    'Ticker',
    'Price',
    'Number Of Share to Buy',
    'One-Year Price Return',
    'One-Year Return Percentile',
    'Six-Month Price Return',
    'Six-Month Return Percentile',
    'Three-Month Price Return',
    'Three-Month Return Percentile',    
    'One-Month Price Return',
    'One-Month Return Percentile',
    'HQM Score'
    
]
hqm_dataframe = pd.DataFrame(columns=hqm_columns)
hqm_dataframe

for symbol_string in symbol_strings:
        batch_api_url = f'https://sandbox.iexapis.com/stable/stock/market/batch?symbols={symbol_string},fb&types=price,stats&token={IEX_CLOUD_API_TOKEN}'
        print(batch_api_url)
        data = requests.get(batch_api_url).json()
        for symbol in symbol_string.split(','):
            try:
                hqm_dataframe = hqm_dataframe.append(
                    pd.Series([
                        symbol,
                        float(data[symbol]['price']),
                        'N/A',
                        data[symbol]['stats']['year1ChangePercent'],
                        'N/A',
                        data[symbol]['stats']['month6ChangePercent'],
                        'N/A',
                        data[symbol]['stats']['month3ChangePercent'],
                        'N/A',    
                        data[symbol]['stats']['month1ChangePercent'],
                        'N/A',
                        'N/A'
                    ],index=hqm_columns                   
                    ),ignore_index=True
                )
            except:
                print("This Symbol having issue while fetching data->"+str(symbol))
hqm_dataframe

https://sandbox.iexapis.com/stable/stock/market/batch?symbols=A,AAL,AAP,AAPL,ABBV,ABC,ABMD,ABT,ACN,ADBE,ADI,ADM,ADP,ADSK,AEE,AEP,AES,AFL,AIG,AIV,AIZ,AJG,AKAM,ALB,ALGN,ALK,ALL,ALLE,ALXN,AMAT,AMCR,AMD,AME,AMGN,AMP,AMT,AMZN,ANET,ANSS,ANTM,AON,AOS,APA,APD,APH,APTV,ARE,ATO,ATVI,AVB,AVGO,AVY,AWK,AXP,AZO,BA,BAC,BAX,BBY,BDX,BEN,BF.B,BIIB,BIO,BK,BKNG,BKR,BLK,BLL,BMY,BR,BRK.B,BSX,BWA,BXP,C,CAG,CAH,CARR,CAT,CB,CBOE,CBRE,CCI,CCL,CDNS,CDW,CE,CERN,CF,CFG,CHD,CHRW,CHTR,CI,CINF,CL,CLX,CMA,CMCSA,fb&types=price,stats&token=Tpk_059b97af715d417d9f49f50b51b1c448
https://sandbox.iexapis.com/stable/stock/market/batch?symbols=CME,CMG,CMI,CMS,CNC,CNP,COF,COG,COO,COP,COST,COTY,CPB,CPRT,CRM,CSCO,CSX,CTAS,CTL,CTSH,CTVA,CTXS,CVS,CVX,CXO,D,DAL,DD,DE,DFS,DG,DGX,DHI,DHR,DIS,DISCA,DISCK,DISH,DLR,DLTR,DOV,DOW,DPZ,DRE,DRI,DTE,DUK,DVA,DVN,DXC,DXCM,EA,EBAY,ECL,ED,EFX,EIX,EL,EMN,EMR,EOG,EQIX,EQR,ES,ESS,ETFC,ETN,ETR,EVRG,EW,EXC,EXPD,EXPE,EXR,F,FANG,FAST,FB,FBHS,FCX,FDX,FE,FFIV,FIS,FISV,FITB,FLIR,FLS,FLT,FMC,FOX,FOXA,FRC,FRT

Ticker    Price Number Of Share to Buy One-Year Price Return  \
0        A  143.870                    N/A              0.157306   
1      AAL   18.180                    N/A             -0.044341   
2      AAP  226.860                    N/A              0.470495   
3     AAPL  176.400                    N/A              0.279484   
4     ABBV  146.330                    N/A              0.356301   
..     ...      ...                    ...                   ...   
499    YUM  132.350                    N/A              0.217969   
500    ZBH  116.954                    N/A             -0.234549   
501   ZBRA  510.060                    N/A              0.238904   
502   ZION   71.690                    N/A              0.547192   
503    ZTS  207.420                    N/A              0.261108   

    One-Year Return Percentile Six-Month Price Return  \
0                          N/A              -0.094053   
1                          N/A              -0.224009   
2                          N/A               0.114204   
3                          N/A               0.189069   
4                          N/A                 0.2682   
..                         ...                    ...   
499                        N/A              -0.058419   
500                        N/A              -0.180057   
501                        N/A              -0.108972   
502                        N/A                 0.3114   
503                        N/A              -0.009014   

    Six-Month Return Percentile Three-Month Price Return  \
0                           N/A                -0.098593   
1                           N/A                -0.257379   
2                           N/A                -0.007778   
3                           N/A                 0.143243   
4                           N/A                  0.21979   
..                          ...                      ...   
499                         N/A                -0.007535   
500                         N/A                -0.137951   
501                         N/A                -0.176726   
502                         N/A                 0.108503   
503                         N/A                -0.078509   

    Three-Month Return Percentile One-Month Price Return  \
0                             N/A              -0.053778   
1                             N/A              -0.119558   
2                             N/A              -0.061751   
3                             N/A               0.003594   
4                             N/A               0.052529   
..                            ...                    ...   
499                           N/A              -0.087242   
500                           N/A              -0.049435   
501                           N/A              -0.096553   
502                           N/A               0.033958   
503                           N/A              -0.083986   

    One-Month Return Percentile HQM Score  
0                           N/A       N/A  
1                           N/A       N/A  
2                           N/A       N/A  
3                           N/A       N/A  
4                           N/A       N/A  
..                          ...       ...  
499                         N/A       N/A  
500                         N/A       N/A  
501                         N/A       N/A  
502                         N/A       N/A  
503                         N/A       N/A  

[504 rows x 12 columns]

# Calculating Momemtum Percentile

In [185]:
time_periods = [
    'One-Year',
    'Six-Month',
    'Three-Month',
    'One-Month'
]
for row in hqm_dataframe.index:
    for time_period in time_periods:
        if hqm_dataframe.loc[row, f'{time_period} Price Return'] == None:
            hqm_dataframe.loc[row, f'{time_period} Price Return'] = 0
hqm_dataframe
list1 = []
for row in hqm_dataframe.index:
    for time_period in time_periods:
        percentile_col = f'{time_period} Return Percentile'
        change_col = f'{time_period} Price Return'
#          print(hqm_dataframe[change_col])
#          print(hqm_dataframe.loc[row,change_col])
        hqm_dataframe.loc[row,percentile_col] = stats.percentileofscore(hqm_dataframe[change_col],hqm_dataframe.loc[row,change_col])

hqm_dataframe



Ticker    Price Number Of Share to Buy One-Year Price Return  \
0        A  143.870                    N/A              0.157306   
1      AAL   18.180                    N/A             -0.044341   
2      AAP  226.860                    N/A              0.470495   
3     AAPL  176.400                    N/A              0.279484   
4     ABBV  146.330                    N/A              0.356301   
..     ...      ...                    ...                   ...   
499    YUM  132.350                    N/A              0.217969   
500    ZBH  116.954                    N/A             -0.234549   
501   ZBRA  510.060                    N/A              0.238904   
502   ZION   71.690                    N/A              0.547192   
503    ZTS  207.420                    N/A              0.261108   

    One-Year Return Percentile Six-Month Price Return  \
0                    48.214286              -0.094053   
1                     15.47619              -0.224009   
2                    86.309524               0.114204   
3                    67.063492               0.189069   
4                    78.373016                 0.2682   
..                         ...                    ...   
499                  57.142857              -0.058419   
500                   3.174603              -0.180057   
501                   60.31746              -0.108972   
502                  92.460317                 0.3114   
503                  63.888889              -0.009014   

    Six-Month Return Percentile Three-Month Price Return  \
0                     23.214286                -0.098593   
1                      4.761905                -0.257379   
2                     75.793651                -0.007778   
3                     85.515873                 0.143243   
4                     92.460317                  0.21979   
..                          ...                      ...   
499                   30.753968                -0.007535   
500                    9.722222                -0.137951   
501                   20.238095                -0.176726   
502                   94.444444                 0.108503   
503                    44.84127                -0.078509   

    Three-Month Return Percentile One-Month Price Return  \
0                       25.396825              -0.053778   
1                         2.97619              -0.119558   
2                        55.15873              -0.061751   
3                       91.865079               0.003594   
4                       97.222222               0.052529   
..                            ...                    ...   
499                     55.357143              -0.087242   
500                     14.880952              -0.049435   
501                      9.722222              -0.096553   
502                     86.111111               0.033958   
503                      32.34127              -0.083986   

    One-Month Return Percentile HQM Score  
0                      42.65873       N/A  
1                          12.5       N/A  
2                      39.68254       N/A  
3                     72.619048       N/A  
4                     88.888889       N/A  
..                          ...       ...  
499                        25.0       N/A  
500                   44.642857       N/A  
501                   20.238095       N/A  
502                   83.134921       N/A  
503                   26.984127       N/A  

[504 rows x 12 columns]

# Calcuating HQM Score

HQM Score is mean of all Percentile 

Percentile --> How many stocks has price return less that price return

Example

0	A	143.870	N/A	0.157306	48.214286	
1	AAL	18.180	N/A	-0.044341	15.47619	
For AAL  has 15 Company has less than -0.44 price return

In [194]:
from statistics import mean

for row in hqm_dataframe.index[:1]:
    momemtum_percentile = []
    for time_period in time_periods:
        momemtum_percentile.append(hqm_dataframe.loc[row,f'{time_period} Return Percentile'])
    hqm_dataframe.loc[row,'HQM Score'] = mean(momemtum_percentile)
    
hqm_dataframe

Ticker    Price Number Of Share to Buy One-Year Price Return  \
0        A  143.870                    N/A              0.157306   
1      AAL   18.180                    N/A             -0.044341   
2      AAP  226.860                    N/A              0.470495   
3     AAPL  176.400                    N/A              0.279484   
4     ABBV  146.330                    N/A              0.356301   
..     ...      ...                    ...                   ...   
499    YUM  132.350                    N/A              0.217969   
500    ZBH  116.954                    N/A             -0.234549   
501   ZBRA  510.060                    N/A              0.238904   
502   ZION   71.690                    N/A              0.547192   
503    ZTS  207.420                    N/A              0.261108   

    One-Year Return Percentile Six-Month Price Return  \
0                    48.214286              -0.094053   
1                     15.47619              -0.224009   
2                    86.309524               0.114204   
3                    67.063492               0.189069   
4                    78.373016                 0.2682   
..                         ...                    ...   
499                  57.142857              -0.058419   
500                   3.174603              -0.180057   
501                   60.31746              -0.108972   
502                  92.460317                 0.3114   
503                  63.888889              -0.009014   

    Six-Month Return Percentile Three-Month Price Return  \
0                     23.214286                -0.098593   
1                      4.761905                -0.257379   
2                     75.793651                -0.007778   
3                     85.515873                 0.143243   
4                     92.460317                  0.21979   
..                          ...                      ...   
499                   30.753968                -0.007535   
500                    9.722222                -0.137951   
501                   20.238095                -0.176726   
502                   94.444444                 0.108503   
503                    44.84127                -0.078509   

    Three-Month Return Percentile One-Month Price Return  \
0                       25.396825              -0.053778   
1                         2.97619              -0.119558   
2                        55.15873              -0.061751   
3                       91.865079               0.003594   
4                       97.222222               0.052529   
..                            ...                    ...   
499                     55.357143              -0.087242   
500                     14.880952              -0.049435   
501                      9.722222              -0.096553   
502                     86.111111               0.033958   
503                      32.34127              -0.083986   

    One-Month Return Percentile  HQM Score  
0                      42.65873  34.871032  
1                          12.5   8.928571  
2                      39.68254  64.236111  
3                     72.619048  79.265873  
4                     88.888889  89.236111  
..                          ...        ...  
499                        25.0  42.063492  
500                   44.642857  18.105159  
501                   20.238095  27.628968  
502                   83.134921  89.037698  
503                   26.984127  42.013889  

[504 rows x 12 columns]

# Selecting 50 best Momentum Stocks

In [203]:
hqm_dataframe.sort_values('HQM Score',ascending=False,inplace=True)
hqm_dataframe = hqm_dataframe[:50]
hqm_dataframe.reset_index(inplace=True)
hqm_dataframe

level_0  index Ticker     Price Number Of Share to Buy  \
0         0    317    MRO    22.641                    N/A   
1         1    160    EOG   119.730                    N/A   
2         2    109    COP    94.030                    N/A   
3         3    212    HAL    33.200                    N/A   
4         4    357    OXY    41.243                    N/A   
5         5    495    XOM    85.534                    N/A   
6         6    148    DVN    54.210                    N/A   
7         7    386    PXD   236.840                    N/A   
8         8    275     LB    83.760                    N/A   
9         9     42    APA    35.010                    N/A   
10       10     89     CF    76.450                    N/A   
11       11    315    MPC    79.950                    N/A   
12       12    175   FANG   138.500                    N/A   
13       13    314    MOS    44.670                    N/A   
14       14    123    CVX   143.610                    N/A   
15       15    218    HES    99.170                    N/A   
16       16    412    SLB    39.801                    N/A   
17       17    122    CVS   113.360                    N/A   
18       18    300    MCK   275.620                    N/A   
19       19    468    VLO    91.660                    N/A   
20       20     11    ADM    78.920                    N/A   
21       21    170    EXC    44.220                    N/A   
22       22    360   PBCT    20.823                    N/A   
23       23    107    COG    22.900                    N/A   
24       24    482    WFC    58.950                    N/A   
25       25    396    RJF   116.980                    N/A   
26       26    407   SCHW    96.792                    N/A   
27       27    485    WMB    31.564                    N/A   
28       28      4   ABBV   146.330                    N/A   
29       29    487    WRB    93.930                    N/A   
30       30    502   ZION    71.690                    N/A   
31       31     98    CMA   100.010                    N/A   
32       32     17    AFL    67.523                    N/A   
33       33    259   JNPR    36.453                    N/A   
34       34    322    MTB   180.637                    N/A   
35       35    262    KEY    27.102                    N/A   
36       36    383    PSX    90.850                    N/A   
37       37    333     NI    29.096                    N/A   
38       38     90    CFG    55.900                    N/A   
39       39    494   XLNX   215.050                    N/A   
40       40    381    PRU   120.500                    N/A   
41       41    231    HSY   208.550                    N/A   
42       42    382    PSA   364.210                    N/A   
43       43    104    CNC    82.130                    N/A   
44       44    100    CME   249.900                    N/A   
45       45    461    UPS   234.854                    N/A   
46       46    458    UNH   497.184                    N/A   
47       47     39   ANTM   458.240                    N/A   
48       48    219    HFC    38.300                    N/A   
49       49     54    AZO  2097.230                    N/A   

   One-Year Price Return One-Year Return Percentile Six-Month Price Return  \
0                1.82309                  99.603175               0.816227   
1               1.152425                  99.007937               0.688682   
2               1.174209                  99.206349               0.666682   
3               0.706786                  95.833333               0.605104   
4               0.822716                   97.81746               0.547276   
5               0.762905                  96.626984               0.454947   
6               2.103423                  99.801587               1.069733   
7               0.888431                  98.214286               0.566659   
8               2.325138                      100.0               0.845667   
9               0.803107                  97.420635          

# Calculating the number of Share to Buy